Change compression fn to save empty metadata dir

giorgio-mariani opened the pull request:
This commit fixes an issue when storing checkpoints with no metadata.
Now the metadata folder is stored in the compressed checkpoint even
if there are no files inside. This avoids an error when trying to load
the checkpoint.

Flegyas reviewed (changes requested):
Thank you for the PR, it is much appreciated!

Can I be the (**really**) fussy one and ask to change a couple of things?

1. PyCharm can be annoying reminding of the built-in shadowing with `dir` as variable name;
2. I think that we used everywhere the operator `/` for joining paths instead of calling the `join` function. Would you mind changing it in the name of consistency and conciseness?

Thanks again, it's great to see people using our work!

giorgio-mariani commented:
> Thank you for the PR, it is much appreciated!
> 
> Can I be the (**really**) fussy one and ask to change a couple of things?
> 
> 1. PyCharm can be annoying reminding of the built-in shadowing with `dir` as variable name;
> 2. I think that we used everywhere the operator `/` for joining paths instead of calling the `join` function. Would you mind changing it in the name of consistency and conciseness?
> 
> Thanks again, it's great to see people using our work!

No problem, thanks for the feedback! 
I have addressed the issues you pointed out in my latest commit, so now everything should be fine.
Let me know if there is something else that should be changed
